### PR TITLE
Support float16 to some operations

### PR DESCRIPTION
**Pull request by kakao-steve-ai**
## 🙏 Describe the pull request

MaxPool2d and AdaptiveAvgPool2d support float16.

## ✅ Checklist

- [x] Code follows the project's coding conventions and style.
- [x] Tests have been added or updated to cover the changes.
- [x] Documentation has been updated, if necessary.

### CODE REVIEW

**Comment by kakao-steve-ai** [comment]
테스트들 중에 input2d, input3d, input4d, target을 사용하지 않고, 함수 안에서 새로 텐서를 생성해서 사용하는 경우도 있는데 이런 경우는 어떻게 커버할지 고민중입니다.

File: tests/conftest.py [hunk 1]
```diff
@@ -16,24 +16,29 @@
...
 
+@pytest.fixture(scope='session', params=[torch.float32, torch.float16])
+def dtype(request):
+    return request.param
+
+
 @pytest.fixture(scope='session')
-def input2d():
-    return torch.randn(512, 512, device='cuda', requires_grad=True)
+def input2d(dtype):
```
Comment (daemyung) [inline tests/conftest.py]: `input` 변수들은 나중에 없어질 예정입니다. `input` 변수를 수정하지 말고 테스트 케이스에서 `dtype`을 받아서 텐서를 생성하세요.

File: trident/operation/softmax.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ def forward(*args, **kwargs):
         assert dim == 1
 
         m, n = x.shape
+        dt = x.dtype
```
Comment (daemyung) [inline trident/operation/softmax.py]: 변수 이름은 `dt`보다 `dtype`이 적합해 보입니다.

File: trident/operation/softmax.py [hunk 3]
```diff
@@ -49,9 +50,15 @@ def get_num_warps():
 
         assert y.is_cuda and x.is_contiguous
 
+        if dt == torch.float16:
```
Comment (daemyung) [inline trident/operation/softmax.py]: 이러한 형변환은 커널에서 실행 되어야 합니다. `Softmax`는 나중에 Triton에 익숙하시고 변경하는게 좋을 것 같습니다.

**Review by daemyung — CHANGES_REQUESTED** [review]

**Comment by daemyung** [comment]
@rbbang 커밋 제목으로 작성한 내용은 커밋 본문으로 이동되는게 더 적합해 보입니다. 커밋 제목은 더 간결하게 작성해 주세요.

File: tests/conftest.py (hunk 1 above)
Comment (kakao-steve-ai) [reply tests/conftest.py]: 네 수정하였습니다.

File: trident/operation/softmax.py (hunk 2 above)
Comment (kakao-steve-ai) [reply trident/operation/softmax.py]: softmax는 나중에 작업하겠습니다.

File: trident/operation/softmax.py (hunk 3 above)
Comment (kakao-steve-ai) [reply trident/operation/softmax.py]: 네 이 부분 나중에 작업하겠습니다.

**Comment by kakao-steve-ai** [comment]
> @rbbang 커밋 제목으로 작성한 내용은 커밋 본문으로 이동되는게 더 적합해 보입니다. 커밋 제목은 더 간결하게 작성해 주세요.

커밋 제목 간결하게 바꾸고, 본문으로 내용 이동시켰습니다.

**Review by daemyung — APPROVED** [review]
LGTM
